python/lsst/obs/base/makeRawVisitInfo.py

```diff
@@ -121,6 +122,28 @@ def getDateAvg(self, md, exposureTime):
         """
         raise NotImplementedError()
 
+    def getDarkTime(self, argDict):
+        """Get the darkTime from the DARKTIME keyword, else expTime, else NaN
+
+        Subclasses should call this function if desired, by putting:
+        argDict['darkTime'] = self.getDarkTime(argDict)
+        in their __init__() method of the derived class.
+
+        @param[in] argDict  argDict
+        @return darkTime darkTime, as inferred from the metadata
+
+        """
+        darkTime = argDict.get("darkTime", float("NaN"))
+        if np.isfinite(darkTime):
+            return darkTime
+
+        self.log.info("darkTime is NaN/Inf; using exposureTime")
+        exposureTime = argDict.get("exposureTime", np.nan)
+        if not np.isfinite(exposureTime):
+            raise RuntimeError("Tried to substitute exposureTime for darkTime but it is also Nan/Inf")
```

Comment:
This error should probably say "but it is not available", since NaN is not a valid FITS keyword value, so you'll know why you got NaN.